src/pages/storage/forms/StorageVolumeFormMain.tsx

```diff
@@ -23,14 +23,21 @@ const StorageVolumeFormMain: FC<Props> = ({ formik }) => {
             {...getFormProps(formik, "name")}
             type="text"
             label="Name"
-            required
+            disabled={formik.values.isReadOnly || !formik.values.isCreating}
+            required={formik.values.isCreating}
+            help={
+              formik.values.isCreating
+                ? undefined
+                : "Click the name in the header to rename the instance"
+            }
           />
           <label className="p-form__label" htmlFor="limits_disk">
             Size
           </label>
           <DiskSizeSelector
             value={formik.values.size}
             setMemoryLimit={(val?: string) => formik.setFieldValue("size", val)}
+            disabled={formik.values.isReadOnly}
           />
           <p className="p-form-help-text">
```

Comment:
So we can just show the field if `formik.values.isCreating` is true otherwise we don't

Reply:
In some cases the size is changeable -- i.e. for `custom` typed volumes.

Maybe it is not clear enough that you have to click "Edit" in order to unlock the fields? There was a bug preventing the Edit button to always be visible on the bottom of this form. That is now displayed correctly.

The only field that can never be edited on volumes through the form is the name, but I think it is good to keep it where it was on creation as deactivated with the help text pointing to click in the header. Otherwise it is not clear how to rename.

Reply:
Yeah the name field is fine where it is.